explorer/update.go

```diff
@@ -307,32 +326,41 @@ func revertChainUpdate(tx UpdateTx, cru chain.RevertUpdate, revertedIndex types.
 				SiacoinElement: se,
 			})
 		}
-	})
+	}
 
 	var newSiafundElements, spentSiafundElements []types.SiafundElement
 	var ephemeralSiafundElements []types.SiafundElement
-	cru.ForEachSiafundElement(func(se types.SiafundElement, created, spent bool) {
+	for _, diff := range cru.SiafundElementDiffs() {
+		created, spent, se := diff.Created, diff.Spent, diff.SiafundElement
 		if created && spent {
 			ephemeralSiafundElements = append(ephemeralSiafundElements, se)
-			return
+			continue
 		}
 
 		if spent {
 			newSiafundElements = append(newSiafundElements, se)
 		} else {
 			spentSiafundElements = append(spentSiafundElements, se)
 		}
-	})
+	}
 
 	fceMap := make(map[types.FileContractID]FileContractUpdate)
-	cru.ForEachFileContractElement(func(fce types.FileContractElement, created bool, rev *types.FileContractElement, resolved, valid bool) {
-		fceMap[fce.ID] = FileContractUpdate{
-			FileContractElement: fce,
+	for _, diff := range cru.FileContractElementDiffs() {
+		var rev *types.FileContractElement
+		if diff.Revision != nil {
+			rev = &types.FileContractElement{
+				ID:           diff.FileContractElement.ID,
+				StateElement: diff.FileContractElement.StateElement,
+				FileContract: *diff.Revision,
+			}
```

Comment:
@lukechampine since this is the second time we need to manually construct the element, how about adding a helper to `core`?